js/web/lib/wasm/wasm-core-impl.ts

```diff
@@ -256,6 +271,10 @@ export const createSession = async(
       outputNames.push(nameString);
 
       if (!BUILD_DEFS.DISABLE_WEBGPU) {
+        if (graphCaptureEnabled) {
+          outputPreferredLocations.push('gpu-buffer');
+          continue;
+        }
```

Comment:
per our previous discussion, we don't want to support this. graph capture should not work with outputPreferredOutputLocation set. we need user to specify a pre-allocated gpu-buffer as output.

So I think we don't need this `graphCaptureEnabled` and `currentEpName`.

Reply:
I understand that `graphCaptureEnabled` is saved for later use. It's used in `run()` to (1)check IO binding state, (2)report error before Run and (3)decide whether to clear output bound config after Run.

I try to avoid `currentEPName`, which introduce states to this file. not sure if there is a good way to do this.

Reply:
At some point, I think `currentEPName` is better than `BUILD_DEFS.DISABLE_WEBGPU`. It seems that `BUILD_DEFS.DISABLE_WEBGPU` is fixed. But the user may change the EP dynamically.  `currentEPName` can reflect what kind of EP the user is using.